src/DeviceInput/Implementations/webDeviceInputSystem.ts

```diff
@@ -224,6 +225,14 @@ export class WebDeviceInputSystem implements IDeviceInputSystem {
      * @param numberOfInputs Number of input entries to create for given device
      */
     private _registerDevice(deviceType: DeviceType, deviceSlot: number, numberOfInputs: number) {
+        if (deviceType === undefined) {
+            throw "Unable to register undefined device";
+        }
+
+        if (deviceSlot === undefined) {
+            throw `Unable to register device ${DeviceType[deviceType]} to undefined slot.`;
+        }

```

Comment:
So I just ran through some tests and deviceType being undefined actually isn't possible BUT you can have an undefined deviceSlot if someone creates a pointerdown event with TouchEvent.  TBH, we can't actually interpret a manually dispatched TouchEvent because it has no pointerId that we could properly extrapolate.  I could remove the deviceType check but I'd like to keep the slot one because if it's undefined, it will use the values for the touch as one level up:
Inputs[DeviceType.Touch][DeviceSlot = undefined] is interpreted as Inputs[DeviceType.Touch]

Reply:
Too late for this PR, but in general If it can actually be undefined, the function signature should probably represent that so that TypeScript can warn you about it.